### DISCUSSION TITLE
buildCreateSlice and lazy slices with typescript

### DISCUSSION THREAD

**Discussion by smartfox87** [reduxjs/redux-toolkit#4366] [Q&A] [answered]
`import { asyncThunkCreator, buildCreateSlice } from "@reduxjs/toolkit";
import type { WithSlice } from "@reduxjs/toolkit";
import { handleRejected } from "@/helpers/processExtraReducersCases.js";
import { rootReducer } from "@/store/index";
import { AccountItem, AccountsList } from "@/types/accounts";
import { createInitialAccountsApi, getAccountsListApi } from "@/api/accounts";

const createAppSlice = buildCreateSlice({
  creators: { asyncThunk: asyncThunkCreator },
});

interface State {
  accountsList: AccountsList | [];
  accountItem: AccountItem | null;
}

const initialState: State = {
  accountsList: [],
  accountItem: null,
};

export const accountsSlice = createAppSlice({
  name: "accounts",
  initialState,
  reducers: (create) => ({
    getAccountsListThunk: create.asyncThunk<AccountsList, never, { rejectValue: string }>(
      async (_, { rejectWithValue }) => {
        const { data, error } = await getAccountsListApi();
        if (error) throw rejectWithValue(error.message);
        if (!data?.length) {
          const { data, error } = await createInitialAccountsApi();
          if (error) throw rejectWithValue(error.message);
          return data;
        }
        return data;
      },
      {
        rejected: handleRejected,
        fulfilled: (state, { payload }) => {
          state.accountsList = payload;
        },
      },
    ),
  }),
});

declare module "@/store/index" {
  export interface LazyLoadedSlices extends WithSlice<typeof accountsSlice> {}
}

const injectedReducers = rootReducer.inject(accountsSlice);

export const { getAccountsListThunk } = accountsSlice.actions;`
**I get error:** 

`TS2322: Type
(create: ReducerCreators<State>) => {   getAccountsListThunk: AsyncThunkSliceReducerDefinition<State, never, AccountsList, {     rejectValue: string;   }>; }
is not assignable to type
ValidateSliceCaseReducers<State, SliceCaseReducers<State>> | ((creators: ReducerCreators<State>) => SliceCaseReducers<...>)
Type
(create: ReducerCreators<State>) => {   getAccountsListThunk: AsyncThunkSliceReducerDefinition<State, never, AccountsList, {     rejectValue: string;   }>; }
is not assignable to type
(creators: ReducerCreators<State>) => SliceCaseReducers<State>
Type
{   getAccountsListThunk: AsyncThunkSliceReducerDefinition<State, never, AccountsList, {     rejectValue: string;   }>; }
is not assignable to type SliceCaseReducers<State>
Type '{ getAccountsListThunk: AsyncThunkSliceReducerDefinition<State, never, AccountsList, { rejectValue: string; }>; }' is not assignable to type 'Record<string, CaseReducer<State, { payload: any; type: string; }> | CaseReducerWithPrepare<State, PayloadAction<any, string, any, any>>>'.
Property getAccountsListThunk is incompatible with index signature.
Type 'AsyncThunkSliceReducerDefinition<State, never, AccountsList, { rejectValue: string; }>' is not assignable to type 'CaseReducer<State, { payload: any; type: string; }> | CaseReducerWithPrepare<State, PayloadAction<any, string, any, any>>'.
createSlice.d.ts(138, 5): The expected type comes from property reducers which is declared here on type
CreateSliceOptions<State, SliceCaseReducers<State>, "accounts", "accounts", SliceSelectors<State>>`

**How can I make correct typescript code?**

**Comment by EskiMojo14** [comment, marked as answer]
try using `void` instead of `never` to indicate you don't need an argument

**Reply by smartfox87** [reply]
Thanks a lot. Your solution helped me
